extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubePlaylistLinkHandlerFactory.java

```diff
@@ -67,4 +74,33 @@ public boolean onAcceptUrl(final String url) {
         }
         return true;
     }
+
+    /**
+     * * If it is a mix (auto-generated playlist) URL, return a {@link LinkHandler} where the URL is
+     * like
+     * <code>https://youtube.com/watch?v=videoId&list=playlistId</code>.
+     * <p>Otherwise use super</p>
+     */
+    @Override
+    public ListLinkHandler fromUrl(final String url) throws ParsingException {
+        try {
+            final URL urlObj = Utils.stringToURL(url);
+            final String listID = Utils.getQueryValue(urlObj, "list");
+            if (listID != null && YoutubeParsingHelper.isYoutubeMixId(listID)) {
+                String videoID = Utils.getQueryValue(urlObj, "v");
+                if (videoID == null) {
+                    videoID = listID.substring(2);
```

Comment:
This doesn't work properly for the Mix types having a prefix longer than 2 letters.